test/visual/globalCompositeOperation.js

```diff
@@ -0,0 +1,149 @@
+
+// based on https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/globalCompositeOperation#operations
+// source https://github.com/mdn/content/blob/main/files/en-us/web/api/canvas_api/tutorial/compositing/example/index.md
+
+/**
+ * @type GlobalCompositeOperation
+ */
+const OPERATIONS = [
+    'color',
+    'color-burn',
+    'color-dodge',
+    'copy',
+    'darken',
+    'destination-atop',
+    'destination-in',
+    'destination-out',
+    'destination-over',
+    'difference',
+    'exclusion',
+    'hard-light',
+    'hue',
+    'lighten',
+    'lighter',
+    'luminosity',
+    'multiply',
+    'overlay',
+    'saturation',
+    'screen',
+    'soft-light',
+    'source-atop',
+    'source-in',
+    'source-out',
+    'source-over',
+    'xor',
+
+];
+
+const radius = 100;
+
+const size = {
+    width: 320,
+    height: 300
+};
+
+/**
+ * source https://github.com/mdn/content/blob/main/files/en-us/web/api/canvas_api/tutorial/compositing/example/index.md#utility-functions
+ */
+const colorSphere = (r) => {
+    const c = fabric.util.createCanvasElement()
+    c.width = c.height = r * 2;
+    const ctx = c.getContext('2d');
+    const rotate = (1 / 360) * Math.PI * 2; // per degree
+    const oleft = 0;
+    const otop = 0;
+    for (let n = 0; n < 360; n++) {
+        const gradient = ctx.createLinearGradient(oleft + r, otop, oleft + r, otop + r);
+        gradient.addColorStop(0, "rgba(0,0,0,0)");
+        gradient.addColorStop(0.7, `hsl(${(n + 300) % 360},100%,50%)`);
+        gradient.addColorStop(1, "rgba(255,255,255,1)");
+        ctx.beginPath();
+        ctx.moveTo(oleft + r, otop);
+        ctx.lineTo(oleft + r, otop + r);
+        ctx.lineTo(oleft + r + 6, otop);
+        ctx.fillStyle = gradient;
+        ctx.fill();
+        ctx.translate(oleft + r, otop + r);
+        ctx.rotate(rotate);
+        ctx.translate(-(oleft + r), -(otop + r));
+    }
+    return c;
+};
```

Comment:
for the sake of tests stability, should we make an image of this and load it once at tests start?
Here we are testing low level canvas code, that under node could break.

Reply:
done